swaybar/render.c

```diff
@@ -260,15 +290,23 @@ static uint32_t render_status_line(cairo_t *cairo,
 }
 
 static uint32_t render_binding_mode_indicator(cairo_t *cairo,
-		struct swaybar_config *config, const char *mode, double x,
-		uint32_t height) {
+		struct swaybar_output *output, struct swaybar_config *config,
+		const char *mode, double x, uint32_t height) {
+
+	height *= output->scale;
+
 	int text_width, text_height;
 	get_text_size(cairo, config->font, &text_width, &text_height,
-			1, true, "%s", mode);
+			output->scale, true, "%s", mode);
+
+	int ws_vertical_padding = WS_VERTICAL_PADDING * output->scale;
+	int ws_horizontal_padding = WS_HORIZONTAL_PADDING * output->scale;
+	int border_width = BORDER_WIDTH * output->scale;
+
 	uint32_t ideal_height = text_height + ws_vertical_padding * 2
 		+ border_width * 2;
-	if (height < ideal_height) {
-		return ideal_height;
+	if (height < ideal_height / output->scale) {
```

Comment:
Here's an example of why this PR is unreadable: here you compare `height` (in buffer coordinates) with `ideal_height / output->scale` (in surface coordinates). That doesn't make sense.